cmd/thanos/tools_bucket.go

```diff
@@ -414,19 +414,17 @@ func registerBucketWeb(m map[string]setupFunc, root *kingpin.CmdClause, name str
 // Provide a list of resolution, can not use Enum directly, since string does not implement int64 function.
 func listResLevel() []string {
 	return []string{
-		strconv.FormatInt(downsample.ResLevel0, 10),
-		strconv.FormatInt(downsample.ResLevel1, 10),
-		strconv.FormatInt(downsample.ResLevel2, 10)}
+		time.Duration(downsample.ResLevel0).String(),
+		time.Duration(downsample.ResLevel1).String(),
+		time.Duration(downsample.ResLevel2).String()}
 }
 
 func registerBucketReplicate(m map[string]setupFunc, root *kingpin.CmdClause, name string, objStoreConfig *extflag.PathOrContent) {
 	cmd := root.Command("replicate", fmt.Sprintf("Replicate data from one object storage to another. NOTE: Currently it works only with Thanos blocks (%v has to have Thanos metadata).", block.MetaFilename))
 	httpBindAddr, httpGracePeriod := regHTTPFlags(cmd)
 	toObjStoreConfig := regCommonObjStoreFlags(cmd, "-to", false, "The object storage which replicate data to.")
-	// TODO(bwplotka): Allow to replicate many resolution levels.
-	resolution := cmd.Flag("resolution", "Only blocks with this resolution will be replicated.").Default(strconv.FormatInt(downsample.ResLevel0, 10)).HintAction(listResLevel).Int64()
-	// TODO(bwplotka): Allow to replicate many compaction levels.
-	compaction := cmd.Flag("compaction", "Only blocks with this compaction level will be replicated.").Default("1").Int()
+	resolutions := cmd.Flag("resolution", "Only blocks with these resolutions will be replicated. Repeated flag.").Default("0s", "5m", "1h").HintAction(listResLevel).DurationList()
+	compactions := cmd.Flag("compaction", "Only blocks with these compaction levels will be replicated. Repeated flag.").Default("1", "2", "3", "4").Ints()
```

Comment:
@bwplotka I eventually took a stab and changed the defaults and the format of `--resolution` flag to Go duration.
But if you are against it I'll be happy to roll it back it just felt bit more user friendly.

Reply:
hm, I think that's ok (: